### ISSUE TITLE
[CI] RestrictedTrustManagerTests.testThatDelegateTrustManagerIsRespected FIPS Zulu8 failure

### ISSUE THREAD

**Issue by ywangd** [elastic/elasticsearch#62281]
**Build scan**:
* https://gradle-enterprise.elastic.co/s/hsssjrq5nvkyc
* https://gradle-enterprise.elastic.co/s/slhiqge5ru44k

**Repro line**:
> ./gradlew ':x-pack:plugin:core:test' --tests "org.elasticsearch.xpack.core.ssl.RestrictedTrustManagerTests.testThatDelegateTrustManagerIsRespected" -Dtests.seed=889BBA34D7902844 -Dtests.security.manager=true -Dtests.locale=da -Dtests.timezone=America/Indiana/Petersburg -Druntime.java=8 -Dtests.fips.enabled=true

**Reproduces locally?**:
No

**Applicable branches**:
`7.x`, `7.9` FIPS with Zulu8 JVM

**Failure history**:
Started fail [recently](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-60d,mode:quick,to:now))&_a=(columns:!(_source),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:testThatDelegateTrustManagerIsRespected),sort:!(process.time-start,desc))) on 2020-09-05.

**Failure excerpt**:
```
java.lang.AssertionError: |  
  | Expected: matches pattern PKIX path building failed.* |  
  | but: was "unable to process certificates: Unable to find certificate chain."


at __randomizedtesting.SeedInfo.seed([889BBA34D7902844:4EEEBE7B16C90D2F]:0) |  
-- | --
  |   | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  |   | at org.junit.Assert.assertThat(Assert.java:956) |  
  |   | at org.junit.Assert.assertThat(Assert.java:923) |  
  |   | at org.elasticsearch.xpack.core.ssl.RestrictedTrustManagerTests.assertNotValid(RestrictedTrustManagerTests.java:180) |  
  |   | at org.elasticsearch.xpack.core.ssl.RestrictedTrustManagerTests.testThatDelegateTrustManagerIsRespected(RestrictedTrustManagerTests.java:138) |  
  |   | at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) |  
  |   | at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) |  
  |   | at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) |  
  |   | at java.lang.reflect.Method.invoke(Method.java:498) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49) |  
  |   | at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45) |  
  |   | at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48) |  
  |   | at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64) |  
  |   | at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368) |  
  |   | at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817) |  
  |   | at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894) |  
  |   | at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41) |  
  |   | at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) |  
  |   | at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53) |  
  |   | at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47) |  
  |   | at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64) |  
  |   | at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368) |  
  |   | at java.lang.Thread.run(Thread.java:748)


```

Notes: Original failure report was [filed](https://github.com/elastic/elasticsearch/issues/61316#issuecomment-689287766) under #61316. But it makes more sense to track it separately since they are different categories of failures.

**Comment by nknize** [comment]
Failed again:  https://groups.google.com/a/elastic.co/g/build-elasticsearch/c/B38uE-4miuY

```
15:07:48 REPRODUCE WITH: ./gradlew ':x-pack:plugin:core:test' --tests "org.elasticsearch.xpack.core.ssl.RestrictedTrustManagerTests.testThatDelegateTrustManagerIsRespected" -Dtests.seed=22057B6FCC5ADDB4 -Dtests.security.manager=true -Dtests.locale=be-BY -Dtests.timezone=Pacific/Easter -Druntime.java=8 -Dtests.fips.enabled=true
```
```
15:07:48   2> java.lang.AssertionError: 
15:07:48     Expected: matches pattern PKIX path building failed.*
15:07:48          but: was "unable to process certificates: Unable to find certificate chain."
15:07:48         at __randomizedtesting.SeedInfo.seed([22057B6FCC5ADDB4:E4707F200D03F8DF]:0)
15:07:48         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
15:07:48         at org.junit.Assert.assertThat(Assert.java:956)
15:07:48         at org.junit.Assert.assertThat(Assert.java:923)
15:07:48         at org.elasticsearch.xpack.core.ssl.RestrictedTrustManagerTests.assertNotValid(RestrictedTrustManagerTests.java:180)
15:07:48         at org.elasticsearch.xpack.core.ssl.RestrictedTrustManagerTests.testThatDelegateTrustManagerIsRespected(RestrictedTrustManagerTests.java:138)
```

**Comment by nknize** [comment]
Another on 7.9: https://groups.google.com/a/elastic.co/g/build-elasticsearch/c/fW6975rkwyg/m/fn2bVT1sBAAJ

```
14:22:28 REPRODUCE WITH: ./gradlew ':x-pack:plugin:core:test' --tests "org.elasticsearch.xpack.core.ssl.RestrictedTrustManagerTests.testThatDelegateTrustManagerIsRespected" -Dtests.seed=968B4A3502500BB3 -Dtests.security.manager=true -Dtests.locale=it -Dtests.timezone=Europe/Sarajevo -Druntime.java=8 -Dtests.fips.enabled=true
```

```
14:22:26 org.elasticsearch.xpack.core.ssl.RestrictedTrustManagerTests > testThatDelegateTrustManagerIsRespected FAILED
14:22:26     java.lang.AssertionError: 
14:22:26     Expected: matches pattern PKIX path building failed.*
14:22:26          but: was "unable to process certificates: Unable to find certificate chain."
14:22:26         at __randomizedtesting.SeedInfo.seed([968B4A3502500BB3:50FE4E7AC3092ED8]:0)
14:22:26         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
14:22:26         at org.junit.Assert.assertThat(Assert.java:956)
14:22:26         at org.junit.Assert.assertThat(Assert.java:923)
14:22:26         at org.elasticsearch.xpack.core.ssl.RestrictedTrustManagerTests.assertNotValid(RestrictedTrustManagerTests.java:180)
14:22:26         at org.elasticsearch.xpack.core.ssl.RestrictedTrustManagerTests.testThatDelegateTrustManagerIsRespected(RestrictedTrustManagerTests.java:138)
```

**Comment by cbuescher** [comment]
Two more today,  @elastic/es-security should I mute this test or is there anything special in this test that we want to observe?

**Comment by tvernum** [comment]
Sorry, we've just been a little short staffed. It's going to be a bit of a pain to mute this for the relevant JVM/mode, so I can handle it now.

**Comment by tvernum** [comment]
Fixed in #62436
